src/libstd/io/stdio.rs

```diff
@@ -872,6 +874,93 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that a prompt
+/// is printed before capturing input. This function returns
+/// either success ([`Ok`]) or failure ([`Err`]) containing
+/// the input.
+///
+/// # Note
+///
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// # Panics
+///
+/// This function currently panics if it does not receive input,
+/// or when input is empty (end of file / `ctrl + d`).
```

Comment:
Panicking feels pretty unfortunate -- and perhaps I'm missing something but I don't think it actually does panic? Rather we return an Err for these cases but that seems expected/fine.

Reply:
fixed docs